src/Browscap/Generator/BuildGenerator.php

```diff
@@ -82,8 +113,38 @@ protected function postBuild($createZipFile = true)
 
         $zip->close();
 
-        $this->getLogger()->info('finished creating the zip archive');
+        $this->logger->info('finished creating the zip archive');
+    }
+
+    /**
+     * Sets the flag to collect pattern ids during this build
+     *
+     * @param bool $value
+     */
+    public function setCollectPatternIds(bool $value) : void
+    {
+        $this->collectPatternIds = $value;
+    }
+
+    /**
+     * @param string $directory
+     *
+     * @throws \InvalidArgumentException
+     *
+     * @return string
+     */
+    private function checkDirectoryExists(string $directory) : string
+    {
+        $realDirectory = realpath($directory);
+
+        if (false === $realDirectory) {
+            throw new DirectoryMissingException('The directory "' . $directory . '" does not exist, or we cannot access it');
+        }
+
+        if (!is_dir($realDirectory)) {
+            throw new NoDirectoryException('The path "' . $realDirectory . '" did not resolve to a directory');
```

Comment:
Ah, I see, here's the usage of this.

I'd name this one "NotADirectoryException" or "NotDirectoryException" or something like that. "NoDirectory" could mean the same thing as the one above.

Reply:
renamed